pkg/controller/kieapp/defaults/defaults_test.go

```diff
@@ -623,6 +619,83 @@ func testAMQEnvs(t *testing.T, kieserverEnvs []corev1.EnvVar, amqEnvs []corev1.E
 	}
 }
 
+func createJvmTestObject() api.JvmObject {
+    jvmObject := api.JvmObject{
+        JavaOptsAppend:             "-Dsome.property=foo",
+        JavaMaxMemRatio:            Pint32(50),
+        JavaInitialMemRatio:        Pint32(25),
+        JavaMaxInitialMem:          Pint32(4096),
+        JavaDiagnostics:            Pbool(true),
+        JavaDebug:                  Pbool(true),
+        JavaDebugPort:              Pint32(8787),
+        ContainerCoreLimit:         Pint32(2),
+        ContainerMaxMemory:         Pint32(1024),
+        GcMinHeapFreeRatio:         Pint32(20),
+        GcMaxHeapFreeRatio:         Pint32(40),
+        GcTimeRatio:                Pint32(4),
+        GcAdaptiveSizePolicyWeight: Pint32(90),
+        GcMaxMetaspaceSize:         Pint32(100),
+        GcContainerOptions:         "-XX:+UseG1GC",
+    }
+    return jvmObject
+}
+
+func testJvmEnv(t *testing.T, envs []corev1.EnvVar) {
+	for _, env := range envs {
+		switch e := env.Name; e {
+		case "JAVA_OPTS_APPEND":
+			assert.Equal(t, "-Dsome.property=foo", env.Value)
+
+		case "JAVA_MAX_MEM_RATIO":
+			assert.Equal(t, "50", env.Value)
+
+		case "JAVA_INITIAL_MEM_RATIO":
+			assert.Equal(t, "25", env.Value)
+
+		case "JAVA_MAX_INITIAL_MEM":
+			assert.Equal(t, "4096", env.Value)
+
+		case "JAVA_DIAGNOSTICS":
+			assert.Equal(t, "true", env.Value)
+
+		case "JAVA_DEBUG":
+			assert.Equal(t, "true", env.Value)
+
+		case "JAVA_DEBUG_PORT":
+			assert.Equal(t, "8787", env.Value)
+
+		case "CONTAINER_CORE_LIMIT":
+			assert.Equal(t, "2", env.Value)
+
+		case "CONTAINER_MAX_MEMORY":
+			assert.Equal(t, "1024", env.Value)
+
+		case "GC_MIN_HEAP_FREE_RATIO":
+			assert.Equal(t, "20", env.Value)
+
+		case "GC_MAX_HEAP_FREE_RATIO":
+			assert.Equal(t, "40", env.Value)
+
+		case "GC_TIME_RATIO":
+			assert.Equal(t, "4", env.Value)
+
+		case "GC_ADAPTIVE_SIZE_POLICY_WEIGHT":
+			assert.Equal(t, "90", env.Value)
+
+		case "GC_MAX_METASPACE_SIZE":
+			assert.Equal(t, "100", env.Value)
+
+		/* TODO
```

Comment:
@spolti , @tchughesiv When I run the unit test, the actual value has somehow become html-escaped. I tried to un-escape it using `html.UnescapeString`, to no avail. I'm pretty stumped here why it's doing that. Help?

Reply:
The lines got messed up since I did a force push... Just look for the TODO and see that comment block, please. :)